Merge physbone api (#1597)
anatawa12/AvatarOptimizer: .docs/content/docs/developers/component-api/index.md

For example, enabling the inversion option of `AAO Remove Mesh By Box` component, added in v1.8.0, changes the meaning of box, which makes it incompatible with tools intended only for v1.7 and earlier.

</blockquote>

<blockquote class="book-hint info">

Since Avatar Optimizer v1.9.0, Avatar Optimizer provides Component API for the `MergePhysBone` component, but it is special with semantic versioning.

This component is deeply integrated with the PhysBone component of VRChat SDK.
If the implementation of the PhysBone component is changed, corresponding changes in `MergePhysBone` may be required to maintain functionality.
In that case, we may add new properties or change existing properties of `MergePhysBone` without following semantic versioning.

1. When we add that new backward-compatible properties to `MergePhysBone` for following new PhysBone properties.\

> This is typically done in bump[^vrcsdk-versioning] version of VRChat SDK.\

Reviewer: ```suggestion
This is typically done in bumps[^vrcsdk-versioning] version of VRChat SDK.\
```

For example, enabling the inversion option of `AAO Remove Mesh By Box` component, added in v1.8.0, changes the meaning of box, which makes it incompatible with tools intended only for v1.7 and earlier.

</blockquote>

<blockquote class="book-hint info">

The `MergePhysBone` component, made public API in v1.9.0, is special with semantic versioning.

This component is deeply integrated with the PhysBone component of VRChat SDK.
Therefore, changes in PhysBone component may require corresponding changes in `MergePhysBone` to maintain functionality.

> Therefore, we may add new properties or change existing properties of `MergePhysBone` to align with changes in the PhysBone component.

Reviewer: ```suggestion
This component is deeply integrated with the PhysBone component of VRChat SDK.
If the implementation of the PhysBone component is changed, corresponding changes in `MergePhysBone` may be required to maintain functionality.
In that case, we may add new properties or change existing properties of `MergePhysBone` without following semantic versioning.
```

This is typically done in bump[^vrcsdk-versioning] version of VRChat SDK, and corresponding patch version of Avatar Optimizer.
We treat adding support for new PhysBone properties as "bugfix for unsupported PhysBone features", so this is done in patch version.

The second is that we might change signature of existing properties (in other words, introduce breaking changes) of `MergePhysBone`,
when PhysBone changes existing properties in a breaking way.
Avatar Optimizer declares compatibility with specific range of breaking versions of VRChat SDK,
so this change can only be done when Avatar Optimizer introduces support for new breaking[^vrcsdk-versioning]
version of VRChat SDK.
We work to minimize such breaking changes, but please be aware of this possibility when using `MergePhysBone`.

To protect your code from such breaking changes, we recommend to check the VRChat SDK version in vpm dependency.
As described above, such breaking changes only happen when Avatar Optimizer introduces support for new breaking version of VRChat SDK.
Therefore, if you lock the VRChat SDK version in vpm dependency, your code will be safe from such breaking changes,

> as long as no conflict in package version happens.

Reviewer: ```suggestion
1. When we add that new backward-compatible properties to `MergePhysBone` for following new PhysBone properties.\
This is typically done in bump[^vrcsdk-versioning] version of VRChat SDK.\
We treat adding support for new PhysBone properties as "bugfix for unsupported PhysBone features", so this is done in patch version of Avatar Optimizer.

2. When we change signature of existing properties (in other words, introduce breaking changes) of `MergePhysBone` for following breaking changes of existing PhysBone properties.\
Avatar Optimizer declares compatibility with specific range of breaking versions of VRChat SDK,
so this change can only be done when Avatar Optimizer introduces support for new breaking[^vrcsdk-versioning] version of VRChat SDK.\
We work to minimize such breaking changes, but please be aware of this possibility when using Component API for `MergePhysBone`.\
In some cases, even when using the same version of Avatar Optimizer, property types and others may be different depending on used VRChat SDK version.

To protect your code from breaking changes like above, we recommend to check the VRChat SDK version with vpmDependencies.\
Breaking changes like above only happen when Avatar Optimizer introduces support for new breaking version of VRChat SDK.\
Therefore, if you lock the VRChat SDK version with vpmDependencies, your code will be safe from such breaking changes, as long as no conflict in package version happens.
```
